src/utils/vectors.ts

```diff
@@ -0,0 +1,97 @@
+import { emptyCost, Generator, GeneratorInstance } from '../armature/Generator';
+import { Node } from '../armature/Node';
+
+import { vec4 } from 'gl-matrix';
+import { range } from 'lodash';
+
+const vectorPool: vec4[] = [];
+let nextVectorIndex: number = 0;
+const zero = vec4.fromValues(0, 0, 0, 1);
+
+// Convenience functions to reuse a pool of vectors
+const getNewVector = () => {
+    if (nextVectorIndex < vectorPool.length) {
+        const nextVector = vectorPool[nextVectorIndex];
+        nextVectorIndex += 1;
+
+        return nextVector;
+    } else {
+        const nextVector = vec4.create();
+        vectorPool.push(nextVector);
+        nextVectorIndex += 1;
+
+        return nextVector;
+    }
+};
+
+const done = () => {
+    nextVectorIndex = 0;
+};
+
+/**
+ * Given a generator and a component to generate, creates a set of direction vectors generated as
+ * children of that component.
+ *
+ * Importantly, this reuses vectos in each call, so it assumes the vectors have been read and
+ * processed and can be reused by the time this function is called again.
+ *
+ * @returns {vec4[]} A set of generated, scaled vectors.
+ */
+export function worldSpaceVectors(generator: Generator, start: string): vec4[] {
+    const nodeLocations: Map<Node, vec4> = new Map<Node, vec4>();
+    const vectors: vec4[] = [];
+
+    const instance = new GeneratorInstance(generator, { getCost: () => emptyCost });
+    instance.generate({ start, depth: 0 });
+
+    // Grow ten times, since any more than that starts to have low probabilities and lower weight
+    range(10).forEach(() => {
+        instance.growIfPossible((added: Node[]) => {
```

Comment:
From my understanding, using `growIfPossible` means `worldSpaceVectors` doesn't do what's described on the tin. This function ends up creating a bunch of vectors that can originate from other spawn points in descendant nodes, not just spawn points that are direct children of `start` as I understood this function was supposed to be doing. Is this behaviour what you intended? If so, can you update the documentation around it to be more clear about descendant parent-to-child vectors being OK to return here as well?

Reply:
It is intentional, so I'll update the comment